### ISSUE TITLE
scikit-learn 0.24 breaks pyriemann (`_init_centroids` is now a class method)

### ISSUE THREAD

**Issue by ErikBjare** [alexandrebarachant/pyRiemann#92]
Using scikit-learn `0.24.0rc1`.

I was working on https://github.com/NeuroTechX/eeg-notebooks/pull/32 and CI caught this: https://github.com/NeuroTechX/eeg-notebooks/runs/1498684706

```
Unexpected failing examples:
/home/runner/work/eeg-notebooks/eeg-notebooks/examples/visual_n170/02r__n170_decoding.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/runner/work/eeg-notebooks/eeg-notebooks/examples/visual_n170/02r__n170_decoding.py", line 38, in <module>
    from pyriemann.estimation import ERPCovariances, XdawnCovariances
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/pyriemann/__init__.py", line 6, in <module>
    from . import clustering
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/pyriemann/clustering.py", line 5, in <module>
    from sklearn.cluster._kmeans import _init_centroids
ImportError: cannot import name '_init_centroids'
```

Looks like scikit-learn rewrote their `_kmeans` module, so `_init_centroids` is now a method of the `KMeans` class instead: https://github.com/scikit-learn/scikit-learn/blob/7cb6b8fde70bd12501e85be8102c46b8ca48405f/sklearn/cluster/_kmeans.py#L895

**Comment by ErikBjare** [comment]
Submitted a workaround in #93

**Comment by qbarthelemy** [comment]
This issue seems to duplicate #87.

**Comment by ErikBjare** [comment]
It's similar, but I don't believe it is quite the same.

**Comment by ErikBjare** [comment]
@qbarthelemy After inspecting your PR it looks like it's the same after all, probably just an artifact of us using slightly different versions of the sklearn prereleases. However, your fix looks like it won't work since _init_centroids is a bound class method which takes `self` (and your use of `pragma: no cover` indicates you haven't actually tested it).

After inspecting the new `_init_centroids` it looks like both our solutions won't work, since the new `_init_centroids` has moved some of its required kwargs into the constructor for `KMeans`.

I'll fix this in my PR #93.
